pkg/auth/native_store_test.go

```diff
@@ -0,0 +1,199 @@
+		err := json.NewDecoder(strings.NewReader(inS)).Decode(&c)
+		if err != nil {
+			return []byte("program failed"), errCommandExited
+		}
+		switch c.ServerURL {
+		case validServerAddress, validServerAddress2:
+			return nil, nil
+		default:
+			return []byte("program failed"), errCommandExited
+		}
+	}
+
+	return []byte(fmt.Sprintf("unknown argument %q with %q", m.arg, inS)), errCommandExited
+}
+
+// Input sets the input to send to a remote credentials helper.
+func (m *mockCommand) Input(in io.Reader) {
+	m.input = in
+}
+
+func mockCommandFn(args ...string) client.Program {
+	return &mockCommand{
+		arg: args[0],
+	}
+}
+
+func TestNativeStore_StoreGetErase(t *testing.T) {
+	creds := auth.Credential{
+		Username: validUsername,
+		Password: validPassword,
+	}
+	s := &nativeAuthStore{
+		programFunc: mockCommandFn,
+	}
+
+	// store creds
+	err := s.Store(validServerAddress, creds)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err.Error())
+	}
+
+	// get creds
+	fetchedCreds, err := s.Get(validServerAddress)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err.Error())
+	}
+	if fetchedCreds != creds {
+		t.Fatalf("expected %+v, got %+v", creds, fetchedCreds)
+	}
+
+	// erase creds
+	err = s.Erase(validServerAddress)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err.Error())
+	}
+	fetchedCreds, err = s.Get(validServerAddress)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err.Error())
+	}
+	if fetchedCreds == auth.EmptyCredential {
+		t.Fatalf("expect empty conf, but got: %+v", fetchedCreds)
+	}
+}
+
+func TestNativeStore_StoreIdentityToken(t *testing.T) {
+	creds := auth.Credential{
+		RefreshToken: validIdentityToken,
+	}
+	s := &nativeAuthStore{
+		programFunc: mockCommandFn,
+	}
+
+	// store creds
+	err := s.Store(validServerAddress, creds)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err.Error())
+	}
+
+	// get creds
+	fetchedCreds, err := s.Get(validServerAddress2)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err.Error())
+	}
+	if fetchedCreds.RefreshToken != creds.RefreshToken {
+		t.Fatalf("expected %+v, got %+v", creds, fetchedCreds)
+	}
+}
+
+func TestNativeStore_FailedGet(t *testing.T) {
+	s := &nativeAuthStore{
+		programFunc: mockCommandFn,
+	}
+	_, err := s.Get(invalidServerAddress)
+	if err == nil {
+		t.Fatalf("expected err, got %v", err)
+	}
+}
+
+func TestNativeStore_GetCredentialsStore_LoadConfigFailed(t *testing.T) {
+	loadConfig = func() (*config.File, error) {
+		return nil, fmt.Errorf("loadConfig err")
+	}
+	_, err := GetCredentialsStore(validServerAddress)
+	if err == nil {
+		t.Fatalf("expect error, got nil")
+	}
+}
+
+func TestNativeStore_GetCredentialsStore_NoHelperSet(t *testing.T) {
+	loadConfig = func() (*config.File, error) {
+		return &config.File{}, nil
+	}
+	_, err := GetCredentialsStore(validServerAddress)
+	if err == nil || err.Error() != "could not get the configured credentials store for registry: "+validServerAddress {
+		t.Fatalf("expect error, got nil")
```

Comment:
The error may not be nil. Maybe using `t.Fatalf("Didn't get the expected error, but got: %v", err)` is better.